genesis_core/user_api/iam/dm/models.py

```diff
@@ -16,30 +16,42 @@
 
 import base64
 import datetime
+import enum
 import hashlib
+import os
 import secrets
+import urllib.parse
 import uuid as sys_uuid
 
+from gcl_iam import algorithms
 from gcl_iam import exceptions as iam_e
 from gcl_iam import tokens
 from gcl_sdk.agents.universal.dm import models as ua_models
 import jinja2
+import pyotp
+from restalchemy.common import contexts
+from restalchemy.dm import filters as ra_filters
 from restalchemy.dm import models
 from restalchemy.dm import properties
 from restalchemy.dm import relationships
 from restalchemy.dm import types as ra_types
-from restalchemy.common import contexts
-from restalchemy.dm import filters as ra_filters
+from restalchemy.dm import types_dynamic as ra_types_dynamic
+from restalchemy.storage import exceptions as ra_storage_exceptions
 from restalchemy.storage.sql import orm
-import pyotp
-
 
 from genesis_core.common import constants as c
 from genesis_core.common import utils as u
 from genesis_core.events import payloads as event_payloads
+from genesis_core.secret import constants as secret_c
+from genesis_core.secret.dm import models as secret_models
 from genesis_core.user_api.iam import constants as iam_c
-from genesis_core.user_api.iam.dm import types
 from genesis_core.user_api.iam import exceptions as iam_exceptions
+from genesis_core.user_api.iam.dm import types
+
+
+class KindModelSelectorType(ra_types_dynamic.KindModelSelectorType):
+    def get_kind_types(self):
+        return [self._kind_type_map[k] for k in sorted(self._kind_type_map)]
```

Comment:
Is it used on every token invokation? If it's a hotpath - it would be great to sort only once and prepare all we need in advance.

Reply:
done

Reply:
Is it used on every token invokation?  - only for jwks endpoint